pep-0590.rst

```diff
@@ -48,145 +48,212 @@ The function pointer type
 Calls are made through a function pointer taking the following parameters:
 
 * ``PyObject *callable``: The called object
-* ``Py_ssize_t n``: The number of arguments plus an optional offset flag for the first argument in vector.
+* ``Py_ssize_t n``: The number of arguments plus the optional flag ``PY_VECTORCALL_PREPEND`` (see below)
 * ``PyObject **args``: A vector of arguments
-* ``PyTupleObject *kwnames``: A tuple of the names of the named arguments.
+* ``PyObject *kwnames``: Either ``NULL`` or a non-empty tuple of the names of the keyword arguments
```

Comment:
Why? The protocol is easier to use if there is a unique way to specify "no keyword arguments".

Reply:
I don't have a strong opinion here, but I agree with Jeroen. The caller probably knows already if the tuple is empty or not, so why not help the callee here? As long as `PyObject_VectorCall()` is used for calling into vectorcallees, it can normalise that case internally, and everyone who does not use it is on their own anyway.